tutorials/sphinx-tutorials/multiagent_ppo.py

```diff
@@ -0,0 +1,789 @@
+# and outputs look like. Actions will automatically be drawn at random from the action spec
+# domain.
+#
+n_rollout_steps = 5
+rollout = env.rollout(n_rollout_steps)
+print("rollout of three steps:", rollout)
+print("Shape of the rollout TensorDict:", rollout.batch_size)
+######################################################################
+# We can see that our rollout has ``batch_size`` of ``(num_vmas_envs, n_rollout_steps)``.
+# This means that all the tensors in it will have those leading dimensions.
+#
+# Looking more in depth, we can see that the output tensordict can be divided in the following way:
+#
+# - *In the root* (accessible by running ``rollout.exclude("next")`` ) we will find all the keys that are available
+#   after a reset is called at the first timestep. We can see their evolution through the rollout steps by indexing
+#   the ``n_rollout_steps`` dimension. Among these keys, we will find the ones that are different for each agent
+#   in the ``rollout["agents"]`` tensordict, which will have batch size ``(num_vmas_envs, n_rollout_steps, n_agents)``
+#   signifying that it is storing the additional agent dimension. The ones outside this agent tensordict
+#   will be the shared ones (in this case only done).
+# - *In the next* (accessible by running ``rollout.get("next")`` ). We will find the same structure as the root,
+#   but for keys that are available only after a step.
+#
+# In torchrl the convention is that done and observations will be present in both root and next (as these are
+# available both at reset time and after a step). Action will only be available in root (as there is no action
+# resulting from a step) and reward will only be available in next (as there is no reward at reset time).
+# This structure follows the one in the Sutton and Barto book where root represents data at time :math:`t` and
+# next represents data at time :math:`t+1` of a world step.
+#
+#
+# Render a random rollout
+# ~~~~~~~~~~~~~~~~~~~~~~~
+#
+# If you are on Google Colab, or on a machine with OpenGL and a GUI, you can actually render a random rollout.
+# This will give you an idea of what a random policy will achieve in this task, in order to compare it
+# with the policy you will train yourself!
+#
+# To render a rollout, follow the instructions in the *Render* section at the end of this tutorial
+# and just remove the line ``policy=policy`` from ``env.rollout()`` .
+#
+#
+# Policy
+# ------
+#
+# PPO utilizes a stochastic policy to handle exploration. This means that our
+# neural network will have to output the parameters of a distribution, rather
+# than a single value corresponding to the action taken.
+#
+# As the data is continuous, we use a Tanh-Normal distribution to respect the
+# action space boundaries. TorchRL provides such distribution, and the only
+# thing we need to care about is to build a neural network that outputs the
+# right number of parameters.
```

Comment:
I'm thinking that we should have a custom `TanhNormalActor` or similar where you can just do `TanhNormalActor(network)` and get everything ready to use

Reply:
Yes that can be done for users to optionally use it but here are 2 thoughts:

- we would then have to make a class for each distribution?
- weren't we moving away from wrappers in favor of sequentiality?

Reply:
The wrapper would actually instantiate a TensorDictSequential subclass